src/components/attributes/Attributes.js

```diff
@@ -1,66 +1,65 @@
 import CommonComponents from './CommonComponents';
 
+const DEFAULT_COMPONENTS = ['visible', 'position', 'scale', 'rotation'];
+
 class AddComponent extends React.Component {
+  static propTypes = {
+    entity: React.PropTypes.element.isRequired
+  };
+
+  /**
+   * Add blank component.
+   * If component is instanced, generate an ID.
+   */
   addComponent = () => {
     var entity = this.props.entity;
     var newComponentName = this.refs.select.value;
-    function isComponentInstanced(componentName) {
-      for (var component in entity.components) {
-        if (component.substr(0,component.indexOf('__')) === componentName)
-          return true;
-      }
-      return false;
-    }
 
-    function findNewInstanceId(componentName) {
-      var i = 2;
-      while (entity.components[componentName + '__' + i]) {
-        i++;
-      }
-      return i;
-    }
-
-    if (AFRAME.components[newComponentName].multiple) {
-      if (isComponentInstanced(newComponentName)) {
-        newComponentName = newComponentName + '__' + findNewInstanceId(newComponentName);
-      }
+    if (AFRAME.components[newComponentName].multiple &&
+        isComponentInstanced(entity, newComponentName)) {
+      newComponentName = newComponentName + '__' +
+                         generateComponentInstanceId(entity, newComponentName);
     }
 
     entity.setAttribute(newComponentName, '');
   }
 
-  render() {
-    var entity = this.props.entity;
-    if (!entity) {
-      return <div></div>;
-    }
-    var usedComponents = Object.keys(this.props.entity.components);
+  /**
+   * Component dropdown options.
+   */
+  renderComponentOptions () {
+    const usedComponents = Object.keys(this.props.entity.components);
+    return Object.keys(AFRAME.components)
+      .filter(function (componentName) {
+        return AFRAME.components[componentName].multiple ||
+               usedComponents.indexOf(componentName) === -1;
+      })
+      .sort()
+      .map(function (value) {
+        return <option key={value} value={value}>{value}</option>;
+      });
+  }
+
+  render () {
+    const entity = this.props.entity;
+    if (!entity) { return <div></div>; }
 
     return (
       <Collapsible>
-        <div className="collapsible-header">
+        <div className='collapsible-header'>
           <span>COMPONENTS</span>
         </div>
-        <div className="collapsible-content">
-          <div className="row">
-            <span className="text">Add</span>
-            <span className="value">
-              <select ref="select">
-              {
-                Object.keys(AFRAME.components)
-                  .filter(function(key){return AFRAME.components[key].multiple || usedComponents.indexOf(key)==-1;})
-                  .sort()
-                  .map(function(value) {
-                    return <option key={value} value={value}>{value}</option>;
-                  }.bind(this))
-              }
+        <div className='collapsible-content'>
```

Comment:
I used to use double. But the linter wants single, and makes sense to just go all the way single.